adapter/pkg/synchronizer/apis_fetcher.go

```diff
@@ -161,52 +219,13 @@ func FetchAPIs(id *string, gwLabel []string, c chan SyncAPIResponse, serviceURL
 	if apiUUIDList != nil {
 		req.Header.Set("Content-Type", "application/json")
 	}
-	// Make the request
-	logger.LoggerSync.Debug("Sending the controle plane request")
-	resp, err = client.Do(req)
-	// In the event of a connection error, the error would not be nil, then return the error
-	// If the error is not null, proceed
-	if err != nil {
-		logger.LoggerSync.Errorf("Error occurred while retrieving APIs from API manager: %v", err)
-		respSyncAPI.Err = err
-		respSyncAPI.Resp = nil
-		c <- respSyncAPI
-		return
-	}
-
-	// get the response in the form of a byte slice
-	respBytes, err = ioutil.ReadAll(resp.Body)
-
-	// If the reading response gives an error
-	if err != nil {
-		logger.LoggerSync.Errorf("Error occurred while reading the response: %v", err)
-		respSyncAPI.Err = err
-		respSyncAPI.ErrorCode = resp.StatusCode
-		respSyncAPI.Resp = nil
-		c <- respSyncAPI
-		return
-	}
-	// For successful response, return the byte slice and nil as error
-	if resp.StatusCode == http.StatusOK {
-		respSyncAPI.Err = nil
-		respSyncAPI.Resp = respBytes
-		c <- respSyncAPI
-		return
-	}
-	// If the response is not successful, create a new error with the response and log it and return
-	// Ex: for 401 scenarios, 403 scenarios.
-	logger.LoggerSync.Errorf("Failure response: %v", string(respBytes))
-	respSyncAPI.Err = errors.New(string(respBytes))
-	respSyncAPI.Resp = nil
-	respSyncAPI.ErrorCode = resp.StatusCode
-	c <- respSyncAPI
-	return
+	return req
 }
 
 // RetryFetchingAPIs function keeps retrying to fetch APIs from runtime-artifact endpoint.
-func RetryFetchingAPIs(c chan SyncAPIResponse, serviceURL string, userName string, password string, skipSSL bool,
-	truststoreLocation string, retryInterval time.Duration, data SyncAPIResponse, endpoint string, sendType bool,
-	requestTimeOut time.Duration, queryParamMap map[string]string) {
+func RetryFetchingAPIs(c chan SyncAPIResponse, data SyncAPIResponse, endpoint string, sendType bool,
+	queryParamMap map[string]string) {
+	retryInterval := workerPool.controlPlaneParams.retryInterval
 	go func(d SyncAPIResponse) {
```

Comment:
check the possibility of removing go routine as it is already stays within a go routine.